profiler/profiler.go

```diff
@@ -603,9 +605,20 @@ func initializeConfig(cfg Config) error {
 func pollProfilerService(ctx context.Context, a *agent) {
 	debugLog("Cloud Profiler Go Agent version: %s", version.Repo)
 	debugLog("profiler has started")
+	defer debugLog("profiler has stopped")
+
+	ticker := time.NewTicker(time.Minute)
```

Comment:
The agent can sometimes be asked by the backend to backoff for a long duration. The ticker will keep generating ticks even the agent is in backoff. This is a waste of resources.

Additionally, in case where a single task is being profileed, the same agent will be asked to collect multiple profile types in a duration of one minute. This approach will force the agent to wait one minute before calling createProfile again, which is not expected.

Reply:
> This is a waste of resources

The overhead of generating a tick once per minute is, as far as I can tell, miniscule. Admittedly I am not an expert on the implementation of the ticker or go's internals, so if you think some other mechanism for guaranteeing a maximum iteration rate is more appropriate I'm open to suggestions.

I do think it's critical, however, to have _something_ in `pollProfilerService()` which very simply (without relying on behavior of dependencies or remote APIs) can be shown to have a reasonable iteration rate under any circumstances.

> Additionally, in case where a single task is being profileed, the same agent will be asked to collect multiple profile types in a duration of one minute.

So let's reduce `minimumIteration` to a smaller value? What is the maximum reasonable rate of profile requests? Surely, it is a finite number, with great latitude between "a reasonable rate" and "so fast the CPU usage is unacceptable".